packages/components/pie-modal/src/defs.ts

```diff
@@ -56,6 +56,27 @@ export interface ModalProps extends RTLComponentProps {
      * The size of the modal; this controls how wide it will appear on the page.
      */
     size: typeof sizes[number];
+
+    /**
+     * The leading action configuration for the modal.
+     */
+    leadingAction?: {
```

Comment:
I would potentially create a type for action to reuse between this and the upcoming trailing action

Reply:
Yeah I was thinking this when adding it. This ticket actually suggested working on both tickets at the same time so i was hoping to pick that up next and then change this. Would that be ok if i left it as it is now then add types in the second ticket?